packages/core/services/capabilities.js

```diff
@@ -0,0 +1,194 @@
+const { ControlledContainerMixin, delay } = require('@semapps/ldp');
+const { MIME_TYPES } = require('@semapps/mime-types');
+const { triple, namedNode } = require('@rdfjs/data-model');
+const fetch = require('node-fetch');
+const CAPABILITIES_ROUTE = '/capabilities';
+
+const containerPermissions = {
+  anon: {
+    read: false,
+    write: false,
+    append: false,
+    control: false,
+  },
+  anyUser: {
+    read: false,
+    write: false,
+    append: false,
+    control: false,
+  },
+};
+
+/**
+ * Service to host the capabilities container.
+ *
+ * @type {import('moleculer').ServiceSchema}
+ *
+ */
+const CapabilitiesService = {
+  name: 'capabilities',
+  mixins: [ControlledContainerMixin],
+  settings: {
+    path: CAPABILITIES_ROUTE,
+    excludeFromMirror: true,
+    permissions: containerPermissions,
+    newResourcesPermissions: containerPermissions,
+  },
+  hooks: {
+    before: {
+      /**
+       * Bypass authorization when getting the resource.
+       *
+       * The URI itself is considered the secret. So no more
+       * authorization is necessary at this point.
+       * If we decide to support capabilities with multiple
+       * authorization factors, this will have to change in
+       * the future.
+       */
+      get: (ctx) => {
+        ctx.params.webId = 'system';
+      },
+    },
+  },
+  events: {
+    /**
+     * A new user was registered:
+     * - Add the capabilities container URI to the webId document.
+     * - Add an invite capability to the capabilities container.
+     */
+    async 'auth.registered'(ctx) {
+      const { webId } = ctx.params;
+
+      const capContainerUri = await this.actions.getContainerUri({ webId }, { parentCtx: ctx });
+
+      // Add a reference to the webId document.
+      ctx.call('ldp.resource.patch', {
+        resourceUri: webId,
+        triplesToAdd: [triple(namedNode(webId), namedNode('apods:capabilities'), namedNode(capContainerUri))],
+        webId,
+      });
```

Comment:
> By the way, you cannot use `apods:capabilities` for `namedNode` because this function doesn't understand prefixes.

Thanks for the heads up, maybe it would be helpful to have statically generated URIs to get around using short forms but maintain readability? Inrupt provides a [generator](https://github.com/inrupt/artifact-generator) which might be pretty helpful. Maybe we can have a chat about that :)

Reply:
> My rationale was that it might not be possible for a service to know where the capabilities container is located unless the path is standardized. But I was not aware of standardized paths and more used to WebID document references, like is done with the profile document.

Right now containers path is not standardized but that will be one of the subject of the refactoring. For non-Pod instances, we use the [VOID endpoint](https://semapps.org/docs/middleware/void) to know about the available containers. We could also use the rather simple Solid [TypeIndex](https://github.com/solid/type-indexes) but the spec is not finished yet. Either way, there is no real need at the moment to indicate this information in the WebID document, so yes maybe it's a bit of overengineering.

Reply:
> Thanks for the heads up, maybe it would be helpful to have statically generated URIs to get around using short forms but maintain readability? Inrupt provides a [generator](https://github.com/inrupt/artifact-generator) which might be pretty helpful. Maybe we can have a chat about that :)

I didn't know about that tool. It's a good idea, I've created an issue for that https://github.com/assemblee-virtuelle/semapps/issues/1163